src/evaluating_students/router.js

```diff
@@ -1,14 +1,354 @@
 const express = require('express');
 const router = express.Router();
+let GradeModel = require("./model/grade").GradeModel;
+let StudentModel = require("./model/student").StudentModel;
+let TeacherModel = require("./model/teacher").TeacherModel;
+let TestModel = require("./model/test").TestModel;
+
+
+/**
+ * @swagger
+ *  paths:
+ *      /teachers:
+ *         get:
+ *           summary: Return list of all teachers
+ *           responses:
+ *              '200':
+ *                  description: "List of all teachers"
+ *              '500':
+ *                  description: "Database query failed"
+ *         put:
+ *           summary: Not implemented.
+ *           responses:
+ *              '501':
+ *                  description: "Not implemented"
+ *         post:
+ *           summary: Add teacher to teacher collection
+ *           parameters:
+ *              -   in: body
+ *                  name: teacher
+ *                  description: The teacher to create.
+ *                  schema:
+ *                      type: object
+ *                      required:
+ *                          -   firstName
+ *                          -   lastName
+ *                      properties:
+ *                          firstName:
+ *                              type: string
+ *                          lastName:
+ *                              type: string
+ *           responses:
+ *              '201':
+ *                  description: "[Saved document teacher]"
+ *              '400':
+ *                  description: "Bad request"
+ *              '500':
+ *                  description: "Failed to save document"
+ *         delete:
+ *           summary: Deletes all documents in the teacher collection
+ *           responses:
+ *              '200':
+ *                  description: "[Count of deleted documents]"
+ *              '500':
+ *                  description: "Failed to delete document"
+ */
+router.route('/teachers')
+    .get((req, res) => {
+        TeacherModel.find({})
+            .then((allTeachers) => res.json(allTeachers))
+            .catch(() => res.sendStatus(500))
```

Comment:
Ik zou persoonlijk alle errors doorsturen naar een error handler met 'next'

Reply:
Dat is wel mooier ja, zag er nu niet echt de meerwaarde van in omdat het zo'n kleine service is en kort project is.